### ISSUE TITLE
Running analyzers on same host as image build can be very slow due to image build artifacts

### ISSUE THREAD

**Issue by zhill** [anchore/anchore#22]
Anchore analysis runs on the "familytree" of the image, which is the set of parentIDs of the layers. Typically this is not a 1:1 mapping to layers, but to images referred to in 'FROM' directives. However, on image builds, docker sets it for each layer. Anchore then runs an unpack() and analysis on each individual layer and all content to that point, causing analysis to take a long time and lots of resources.

Example of build behavior on an image built from ubuntu:latest:
```
docker history testimage1
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
17cada8704f7        30 minutes ago      /bin/sh -c #(nop)  ENV somekey=somevalue        0 B
389be87be712        30 minutes ago      /bin/sh -c echo "Hello5" > /root/hello5         7 B
ba8efdb71b82        30 minutes ago      /bin/sh -c echo "Hello4" > /root/hello4         7 B
c487c9eb4267        30 minutes ago      /bin/sh -c echo "Hello3" > /root/hello3         7 B
dfeefcddffec        30 minutes ago      /bin/sh -c echo "Hello2" > /root/hello2         7 B
ea37c65d781a        30 minutes ago      /bin/sh -c echo "Hello1" > /root/hello1         7 B
8b94a7dd04ae        30 minutes ago      /bin/sh -c echo "Hello" > /root/hello           6 B
f49eec89601e        4 weeks ago         /bin/sh -c #(nop)  CMD ["/bin/bash"]            0 B
<missing>           4 weeks ago         /bin/sh -c mkdir -p /run/systemd && echo '...   7 B
<missing>           4 weeks ago         /bin/sh -c sed -i 's/^#\s*\(deb.*universe\...   1.9 kB
<missing>           4 weeks ago         /bin/sh -c rm -rf /var/lib/apt/lists/*          0 B
<missing>           4 weeks ago         /bin/sh -c set -xe   && echo '#!/bin/sh' >...   745 B
<missing>           4 weeks ago         /bin/sh -c #(nop) ADD file:68f83d996c38a09...   129 MB
```
Images pushed to registries with V2 manifests seem to lose the parentId anyway, so the condition is that on build-servers we have too much familytree data and on pulled images we have none.

Example image built from jenkins:latest with no parentIds populated:
``` docker history nightfurys/jendock
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
1a197c741ce8        5 weeks ago         /bin/sh -c #(nop)  USER [jenkins]               0 B
<missing>           5 weeks ago         /bin/sh -c apt-get update && apt-get insta...   120 MB
<missing>           5 weeks ago         /bin/sh -c echo "deb https://apt.dockerpro...   58 B
<missing>           5 weeks ago         /bin/sh -c apt-key adv --keyserver hkp://h...   2.81 kB
<missing>           5 weeks ago         /bin/sh -c apt-get update && apt-get insta...   19.6 MB
<missing>           5 weeks ago         /bin/sh -c #(nop)  USER [root]                  0 B
<missing>           8 weeks ago         /bin/sh -c #(nop) COPY file:2a6a3e16202b8d...   5.96 kB
<missing>           8 weeks ago         /bin/sh -c #(nop) COPY file:93fb511d485dd2...   3.92 kB
<missing>           8 weeks ago         /bin/sh -c #(nop)  ENTRYPOINT ["/bin/tini"...   0 B
<missing>           8 weeks ago         /bin/sh -c #(nop) COPY file:7eec179a0dd3aa...   1.21 kB
<missing>           8 weeks ago         /bin/sh -c #(nop) COPY file:26c3c5818bc876...   5 kB
<missing>           8 weeks ago         /bin/sh -c #(nop)  USER [jenkins]               0 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  ENV COPY_REFERENCE_FILE...   0 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  EXPOSE 50000/tcp             0 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  EXPOSE 8080/tcp              0 B
<missing>           8 weeks ago         |6 JENKINS_SHA=1b65dc498ba7ab1f5cce64200b9...   328 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  ENV JENKINS_UC=https://...   0 B
<missing>           8 weeks ago         |6 JENKINS_SHA=1b65dc498ba7ab1f5cce64200b9...   70.1 MB
<missing>           8 weeks ago         /bin/sh -c #(nop)  ARG JENKINS_URL=https:/...   0 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  ARG JENKINS_SHA=1b65dc4...   0 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  ENV JENKINS_VERSION=2.32.1   0 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  ARG JENKINS_VERSION          0 B
<missing>           8 weeks ago         /bin/sh -c #(nop) COPY file:c629bc0b9ecb5b...   328 B
<missing>           8 weeks ago         |4 gid=1000 group=jenkins uid=1000 user=je...   822 kB
<missing>           8 weeks ago         /bin/sh -c #(nop)  ENV TINI_SHA=0f78709a0e...   0 B
<missing>           8 weeks ago         /bin/sh -c #(nop)  ENV TINI_VERSION=0.13.1      0 B
<missing>           2 months ago        |4 gid=1000 group=jenkins uid=1000 user=je...   0 B
<missing>           2 months ago        /bin/sh -c #(nop)  VOLUME [/var/jenkins_home]   0 B
<missing>           2 months ago        |4 gid=1000 group=jenkins uid=1000 user=je...   335 kB
<missing>           2 months ago        /bin/sh -c #(nop)  ARG gid=1000                 0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ARG uid=1000                 0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ARG group=jenkins            0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ARG user=jenkins             0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ENV JENKINS_SLAVE_AGENT...   0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ENV JENKINS_HOME=/var/j...   0 B
<missing>           2 months ago        /bin/sh -c apt-get update && apt-get insta...   0 B
<missing>           2 months ago        /bin/sh -c /var/lib/dpkg/info/ca-certifica...   418 kB
<missing>           2 months ago        /bin/sh -c set -x  && apt-get update  && a...   352 MB
<missing>           2 months ago        /bin/sh -c #(nop)  ENV CA_CERTIFICATES_JAV...   0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ENV JAVA_DEBIAN_VERSION...   0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ENV JAVA_VERSION=8u111       0 B
<missing>           2 months ago        /bin/sh -c #(nop)  ENV JAVA_HOME=/usr/lib/...   0 B
<missing>           2 months ago        /bin/sh -c {   echo '#!/bin/sh';   echo 's...   87 B
<missing>           2 months ago        /bin/sh -c #(nop)  ENV LANG=C.UTF-8             0 B
<missing>           2 months ago        /bin/sh -c echo 'deb http://deb.debian.org...   55 B
<missing>           2 months ago        /bin/sh -c apt-get update && apt-get insta...   1.29 MB
<missing>           2 months ago        /bin/sh -c apt-get update && apt-get insta...   123 MB
<missing>           2 months ago        /bin/sh -c apt-get update && apt-get insta...   44.3 MB
<missing>           2 months ago        /bin/sh -c #(nop)  CMD ["/bin/bash"]            0 B
<missing>           2 months ago        /bin/sh -c #(nop) ADD file:1d214d2782eaccc...   123 MB```

The anchore family-tree detection fails on that image as well:
```
anchore toolbox --image nightfurys/jendock show-familytree
+--------------+---------------------------+----------------+--------------+
| Image Id     | Current Repo Tags         | Past Repo Tags | Image Type   |
+--------------+---------------------------+----------------+--------------+
| 1a197c741ce8 | nightfurys/jendock:latest |                | Intermediate |
+--------------+---------------------------+----------------+--------------+```

Proposed solution is to disable familytree analysis by default so that only the image is extracted and squashed, not each in family tree.  A flag, '--familytree' would enable the analysis of each image in the tree if desired.

**Comment by zhill** [comment]
Upon further discussion with @nurmi , a better approach is to provide a set of strategies for analyzing the family tree based on expected usage. Disabling it entirely could cause unexpected output in any of the show-*-diff queries. At minimum, should support first and last in the tree and optionally any image that is not an intermediate image should be scanned. Intermediate as determined by anchore analysis with the --imagetype flag set to anything other than 'none'.

**Comment by zhill** [comment]
Fixed by commit 26b0d3d8c694b21b8202fc5e8c40a7a55c93901d
